src/native/managed/cdac/Microsoft.Diagnostics.DataContractReader.Contracts/Contracts/StackWalk/Context/X86/GCInfoDecoding/InfoHdr.cs

```diff
@@ -0,0 +1,643 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace Microsoft.Diagnostics.DataContractReader.Contracts.StackWalkHelpers;
+
+/// <summary>
+/// based on <a href="https://github.com/dotnet/runtime/blob/main/src/coreclr/inc/gcinfotypes.h">src\inc\gcinfotypes.h</a> InfoHdrSmall
+/// </summary>
+public record struct InfoHdr
+{
+    public byte PrologSize { get; set; }
+    public byte EpilogSize { get; set; }
+
+    public byte EpilogCount { get; set; }     // 0 - 7
+    public bool EpilogAtEnd { get; set; }
+    public bool EdiSaved { get; set; }
+    public bool EsiSaved { get; set; }
+    public bool EbxSaved { get; set; }
+    public bool EbpSaved { get; set; }
+
+    public bool EbpFrame { get; set; }
+    public bool Interruptible { get; set; }
+    public bool DoubleAlign { get; set; }
+    public bool Security { get; set; }
+    public bool Handlers { get; set; }
+    public bool LocalAlloc { get; set; }
+    public bool EditAndContinue { get; set; }
+    public bool VarArgs { get; set; }
+
+    public bool ProfCallbacks { get; set; }
+    public bool GenericsContext { get; set; }
+    public bool GenericsContextIsMethodDesc { get; set; }
+    public ReturnKinds ReturnKind { get; set; }
+
+    public ushort ArgCount { get; set; }
+    public uint FrameSize { get; set; }
+    public uint UntrackedCount { get; set; }
+    public uint VarPtrTableSize { get; set; }
+
+    // Fields not set by the initial table encoding
+    public uint GsCookieOffset { get; set; } = 0;
+    public uint SyncStartOffset { get; set; } = 0;
+    public uint SyndEndOffset { get; set; } = 0;
+    public uint RevPInvokeOffset { get; set; } = INVALID_REV_PINVOKE_OFFSET;
+    public uint NoGCRegionCount { get; set; } = 0;
+
+    public bool HasArgTabOffset { get; set; }
+    public uint ArgTabOffset { get; set; }
+    public List<int> Epilogs { get; set; }
```

Comment:
Use `ImmutableArray` here so you don't need to do defensive copying to avoid writing to collections saved in other instances.